data/filter/Utils.ts

```diff
@@ -32,7 +29,8 @@ export function parseFilter(spec: FilterLike): Filter {
     let s = spec as any;
 
     // Degenerate cases
-    if (!s || s instanceof Filter) return s as Filter;
+    if (s instanceof Filter) return s as Filter;
+    if (!s || (isArray(s) && s.length === 0)) return null;
```

Comment:
We could use `isNil(s)` here instead of the falsey check, but given that any other falsey value would be invalid I wasn't sure that would be desirable.  

Also could unwind the empty array handling, but I didn't really see `[] -> empty CompoundFilter` as a feature - seemed more confusing than anything.